tests/ipex/test_modeling.py

```diff
@@ -255,6 +259,37 @@ def test_pipeline(self, model_arch):
         self.assertEqual(pipe.device, model.device)
         self.assertTrue(all("This is a sample" in item["generated_text"] for item in outputs))
 
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_arch": IPEX_PATCHED_SUPPORTED_ARCHITECTURES,
+                "use_cache": [True, False],
+                "num_beams": [1, 4],
+                "batch_size": [1, 4],
+            }
+        )
+    )
+    @unittest.skipIf(is_ipex_version("<=", "2.3.0"), reason="Only ipex version > 2.3.0 supports ipex model patching")
+    def test_ipex_patching(self, test_name, model_arch, use_cache, num_beams, batch_size):
+        model_id = MODEL_NAMES[model_arch]
+        set_seed(SEED)
+        model = IPEXModelForCausalLM.from_pretrained(model_id, export=True)
+        transformers_model = AutoModelForCausalLM.from_pretrained(model_id)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        tokenizer.pad_token = tokenizer.eos_token
+        texts = ["This is a sample"] * batch_size
+        tokens = tokenizer(texts, padding=True, return_tensors="pt")
+        generation_config = GenerationConfig(
+            max_new_tokens=16, num_beams=num_beams, do_sample=False, use_cache=use_cache
+        )
+        outputs = model.generate(**tokens, generation_config=generation_config)
+        with torch.no_grad():
+            transformers_outputs = transformers_model(**tokens)
+
+        self.assertIsInstance(outputs.logits, torch.Tensor)
+        # Compare tensor outputs
+        self.assertTrue(torch.allclose(outputs.logits, transformers_outputs.logits, atol=1e-4))
```

Comment:
this will fail no ? (comparison after `generate` vs one `forward` for the transformers model)
shouldn't it be :
```suggestion
        outputs = model.generate(**tokens, generation_config=generation_config)
        with torch.no_grad():
            transformers_outputs = transformers_model.generate(**tokens, generation_config=generation_config)

        self.assertIsInstance(outputs.logits, torch.Tensor)
        # Compare tensor outputs
        self.assertTrue(torch.allclose(outputs.logits, transformers_outputs.logits, atol=1e-4))
```
It would also make sense to compare both model's output tensor after one forward :
```python
outputs = model(**tokens)
transformers_outputs = transformers_model(**tokens)
```
(can be done for one set of parameters as we don't need to do it for all cases)